pg_search/src/postgres/delete.rs

```diff
@@ -82,6 +88,21 @@ pub extern "C" fn ambulkdelete(
         }
     }
 
+    // Don't merge here, amvacuumcleanup will merge
+    writer
+        .commit(false)
+        .expect("ambulkdelete: commit should succeed");
+
+    if stats.is_null() {
+        stats = unsafe {
+            PgBox::from_pg(
+                pg_sys::palloc0(std::mem::size_of::<pg_sys::IndexBulkDeleteResult>()).cast(),
+            )
+        };
+        stats.pages_deleted = 0;
+    }
+
+    // Acquire cleanup lock, which ensures that no scans have pins on the index
```

Comment:
maybe just expound upon why this is is important and necessary.  You've spent a *ton* of time getting to this point, so more clearly explaining the why will be useful for everyone.